src/query/ts/m3db/convert.go

```diff
@@ -92,22 +120,23 @@ func blockReplicasFromSeriesIterator(seriesIterator encoding.SeriesIterator, ite
 				if err != nil {
 					return nil, err
 				}
+
 				readers[i] = clonedReader
 			}
-			// todo(braskin): pooling
+
 			iter := encoding.NewMultiReaderIterator(iterAlloc, nil)
 			iter.Reset(readers, start, bs)
-
 			inserted := false
-			for i := range blockReplicas {
-				if blockReplicas[i].start.Equal(start) {
+			for i := range replicas {
```

Comment:
replicas is confusing since that has replicas too ? Also, why not just _, replica := range replicas